src/native/profiler.cc

```diff
@@ -103,6 +103,70 @@ nanoseconds_type startup_time;
 // Logger
 std::shared_ptr<spdlog::logger> Profiler::logger = spdlog::basic_logger_mt("basic_logger", "log.txt");
 
+void Profiler::ParseOptions(const char *options) {
+  if (options == NULL) {
+    fprintf(stderr, "Missing options\n");
+    print_usage();
+    exit(1);
+  } else {
+    logger->info("Received options: {}", options);
+  }
+  std::string options_str(options);
+  std::stringstream ss(options_str);
+  std::string item;
+  std::vector<std::string> cmd_line_options;
+  progress_point = new ProgressPoint();
+  progress_point->lineno = -1;
+  progress_point->method_id = nullptr;
+
+  // split underscore delimited line into options
+  // (we can't use semicolon because bash is dumb)
+  while (std::getline(ss, item, '_')) {
+    cmd_line_options.push_back(item);
+  }
```

Comment:
This was always an annoyance of using the tool, and was one of the reasons we decided to add the client / server front-end that was responsible for passing arguments to the profiler. I assume you guys are just invoking the native agent directly? That's fine if so, but we should plan to add some wrapper script that hides this crappy usability if that's the case.

Reply:
We are building `-agentpath` argument automatically, so we have kind of wrapper. But that's the part of specific workflow. It will be neat to have script that will abstract some parts, right in repo, indeed.